### PR TITLE
Show loading and error messages in Analyze view

### PR DESCRIPTION
**Pull request by lewfish**
To test:
 * Go to the analyze view
 * Check that there is a centered spinner that says Loading.
 * Check that results appear after a few seconds
 * Repeat, except right when spinner starts, switch the network to offline. (To switch the network to offline, in Developer Tools, click the smartphone icon in the upper left corner. Then back in the browser, there should be a network dropdown on the top.)
 * Check that an error message with icon shows. 
 * With network back to normal, check that you can go to the Analyze view, wait for results to show, and then go to the modeling view, and back. The results should show immediately.

I've only had time to fix the existing unit tests for the Analyze view. I also was unable to get the new animation for the Analyze view to work due to complicated CSS issues. I'll make issues for these in a minute.

Connects #80

### CODE REVIEW

File: src/mmw/js/src/analyze/views.js [hunk 1]
```diff
@@ -104,6 +116,16 @@ var AnalyzeWindow = Marionette.LayoutView.extend({
...
+var LoadingView = Marionette.ItemView.extend({
+    template: loadingTmpl,
+    className: 'analyze-message-region'
+});
+
+var ErrorView = Marionette.ItemView.extend({
+    template: errorTmpl,
+    className: 'analyze-message-region'
+});
+
```
Comment (caseypt) [inline src/mmw/js/src/analyze/views.js]: Consider consolidating these into one view that expects a model with `message` and `icon-class` attribute.

**Comment by caseypt** [comment]
+1 Looks good. Ran through the testing steps and everything works as described.

File: src/mmw/js/src/analyze/templates/loading.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class="analyze-message-inner">
+    <div>
+        <i class="fa fa-circle-o-notch fa-spin"></i>
+    </div>
+    <div class="analyze-message-text">
+        Loading Data
```
Comment (caseypt) [inline src/mmw/js/src/analyze/templates/loading.html]: I realize this is the message in the mock-ups, but we may want to switch it to something like "Analyzing" or something more descriptive.

**Comment by rajadain** [comment]
+1 works as advertised. Tested the error message as well. I like that the results are not reloaded if they rendered / loaded correctly and one goes back to draw or forward to model and returns to analyze, but they are reloaded if it errored the first time.

I'm investigating the build instability, which actually seems to have been caused by some previous merges.

File: src/mmw/js/src/analyze/views.js [hunk 3]
```diff
@@ -8,20 +8,35 @@ var $ = require('jquery'),
     coreViews = require('../core/views'),
     chart = require('../core/chart'),
     windowTmpl = require('./templates/window.html'),
+    loadingTmpl = require('./templates/loading.html'),
+    errorTmpl = require('./templates/error.html'),
+    headerTmpl = require('./templates/header.html'),
```
Comment (rajadain) [inline src/mmw/js/src/analyze/views.js]: `headerTmpl` is never used and causing jshint issues. Please remove.

File: src/mmw/js/src/analyze/views.js [hunk 4]
```diff
@@ -68,6 +72,14 @@ var AnalyzeWindow = Marionette.LayoutView.extend({
         }));
     },
 
+    showLoadingView: function() {
+        this.detailsRegion.show(new LoadingView());
```
Comment (rajadain) [inline src/mmw/js/src/analyze/views.js]: In accordance with @caseypt's comment on consolidating these views, consider using a pattern similar to one used in geocoding here: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/geocode/views.js#L13 and changing this to something along the lines of:

```javascript
showLoadingView: function() {
    this.detailsRegion.show(new StatusView(ICON_WORKING, MSG_WORKING));
}
```

File: src/mmw/js/src/analyze/views.js (hunk 3 above)
Comment (kdeloach) [inline src/mmw/js/src/analyze/views.js]: General comment, you can always run `./scripts/npm.sh run lint` locally to uncover any JS lint.

Comment (caseypt) [inline src/mmw/js/src/analyze/views.js]: And/or get it wired up in your editor: http://jshint.com/install/

File: src/mmw/js/src/analyze/templates/error.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<div class="analyze-message-inner">
```
Comment (mmcfarland) [inline src/mmw/js/src/analyze/templates/error.html]: Looks like error.html and loading.html are no longer used, and maybe there should be a message.html to replace them?

File: src/mmw/js/src/analyze/views.js [hunk 6]
```diff
@@ -8,20 +8,33 @@ var $ = require('jquery'),
     coreViews = require('../core/views'),
     chart = require('../core/chart'),
     windowTmpl = require('./templates/window.html'),
+    messageTmpl = require('./templates/message.html'),
```
Comment (mmcfarland) [inline src/mmw/js/src/analyze/views.js]: I don't see this module added in this PR.

File: src/mmw/js/src/analyze/templates/error.html (hunk 5 above)
Comment (lewfish) [inline src/mmw/js/src/analyze/templates/error.html]: I forgot to add/rm the files in my last commit. Glad you noticed.

**Comment by mmcfarland** [comment]
Looks sharp, good improvements.

**Comment by mmcfarland** [comment]
+1

**Comment by mmcfarland** [comment]
Anybody else want to take a look or can we merge this?

**Comment by rajadain** [comment]
I'll have one final look

**Comment by lewfish** [comment]
I'm just waiting for it to finish building, but if you want to take a final look that's fine with me.

**Comment by rajadain** [comment]
+1 functionally works very well, code looks great too.

**Pull request merged**